subgraphs/euler-finance/src/mappings/helpers.ts

```diff
@@ -532,45 +640,167 @@ export function updateUsageMetrics(event: ethereum.Event, from: Address, transac
   hourlyMetrics.timestamp = event.block.timestamp;
   hourlyMetrics.hourlyTransactionCount += 1;
 
-  const accountId = from.toHexString();
-  let account = Account.load(accountId);
   const protocol = getOrCreateLendingProtocol();
+  const account = getOrCreateAccount(from, protocol);
   dailyMetrics.totalPoolCount = protocol.totalPoolCount;
-  if (!account) {
-    account = new Account(accountId);
-    account.save();
 
-    protocol.cumulativeUniqueUsers += 1;
-    protocol.save();
-  }
   hourlyMetrics.cumulativeUniqueUsers = protocol.cumulativeUniqueUsers;
   dailyMetrics.cumulativeUniqueUsers = protocol.cumulativeUniqueUsers;
+  dailyMetrics.cumulativeUniqueDepositors = protocol.cumulativeUniqueDepositors;
+  dailyMetrics.cumulativeUniqueBorrowers = protocol.cumulativeUniqueBorrowers;
+  dailyMetrics.cumulativeUniqueLiquidators = protocol.cumulativeUniqueLiquidators;
+  dailyMetrics.cumulativeUniqueLiquidatees = protocol.cumulativeUniqueLiquidatees;
 
-  // Combine the id and the user address to generate a unique user id for the day
-  const dailyActiveAccountId = ActivityType.DAILY + "-" + from.toHexString() + "-" + id.toString();
-  let dailyActiveAccount = ActiveAccount.load(dailyActiveAccountId);
-  if (!dailyActiveAccount) {
-    dailyActiveAccount = new ActiveAccount(dailyActiveAccountId);
-    dailyActiveAccount.save();
+  if (isNewAccountActivity(ActivityType.DAILY, null, account, event.block.timestamp)) {
     dailyMetrics.dailyActiveUsers += 1;
   }
-
-  // create active account for hourlyMetrics
-  const hourlyActiveAccountId = ActivityType.HOURLY + "-" + from.toHexString() + "-" + hour.toString();
-  let hourlyActiveAccount = ActiveAccount.load(hourlyActiveAccountId);
-  if (!hourlyActiveAccount) {
-    hourlyActiveAccount = new ActiveAccount(hourlyActiveAccountId);
-    hourlyActiveAccount.save();
+  if (isNewAccountActivity(ActivityType.HOURLY, null, account, event.block.timestamp)) {
     hourlyMetrics.hourlyActiveUsers += 1;
   }
+  if (isNewAccountActivity(ActivityType.DAILY, transaction, account, event.block.timestamp)) {
+    updateDailyTransactionMetrics(dailyMetrics, transaction, false);
+  }
+  if (transaction == TransactionType.LIQUIDATE) {
+    const liq = eventToLiquidation(event);
```

Comment:
I had the same dilemma myself. Decided to do it this way instead because felt ugly to have a `violator: Address` parameter on a method called `updateUsageMetrics`. It would be an argument that will have no effect if used on non-liquidation events. Changing the way you use the function based on the value of one of its arguments feels odd, not consistent.

Also given that the information is already there (in the `event: Event`), I thought we don't need to pass it again. Although it requires a cast, potentially breaking things.